drivers/i2c/i2c_esp32.c

```diff
@@ -0,0 +1,781 @@
+			      I2C_TRANS_COMPLETE_INT_ENA_M |
+			      I2C_ARBITRATION_LOST_INT_ENA_M;
+
+	/* Activate the device IRQ */
+	irq_enable(config->irq.line);
+
+out:
+	irq_unlock(key);
+
+	return r;
+}
+
+static inline void i2c_esp32_ack_intrs(const struct i2c_esp32_config *config)
+{
+	/* Acknowledge all interrupts */
+	*config->regs.intr_clr = ~0;
+}
+
+
+static inline void i2c_esp32_reset_fifo(const struct i2c_esp32_config *config)
+{
+	/* Writing 1 and then 0 to these bits will reset the I2C fifo */
+	*config->regs.fifo_cfg |= (I2C_TX_FIFO_RST | I2C_RX_FIFO_RST);
+	*config->regs.fifo_cfg &= ~(I2C_TX_FIFO_RST | I2C_RX_FIFO_RST);
+}
+
+static inline void i2c_esp32_prepare(const struct i2c_esp32_config *config)
+{
+	i2c_esp32_ack_intrs(config);
+	i2c_esp32_reset_fifo(config);
+}
+
+static int i2c_esp32_spin_yield(int *counter)
+{
+	*counter = *counter + 1;
+
+	if (*counter > I2C_ESP32_TIMEOUT) {
+		return -ETIMEDOUT;
+	}
+
+	if (*counter > I2C_ESP32_SPIN_THRESHOLD) {
+		k_yield();
+	}
+
+	return 0;
+}
+
+static int i2c_esp32_transmit_and_wait(struct device *dev,
+				       volatile struct i2c_esp32_cmd *wait_cmd)
+{
+	const struct i2c_esp32_config *config = dev->config->config_info;
+	struct i2c_esp32_data *data = dev->driver_data;
+	u32_t status;
+	int counter = 0;
+	int r;
+
+	i2c_esp32_ack_intrs(config);
+
+	/* Start transmission and wait for the ISR to give the semaphore */
+	*config->regs.ctr |= I2C_TRANS_START;
+	if (k_sem_take(&data->fifo_sem, I2C_ESP32_TIMEOUT_MS) < 0) {
+		return -ETIMEDOUT;
+	}
+
+	/* Check for error conditions before waiting for the bus to finish */
+	status = *config->regs.intr_raw;
+	if (status & (I2C_ARBITRATION_LOST_INT_RAW | I2C_ACK_ERR_INT_RAW)) {
+		return -EIO;
+	}
+	if (status & I2C_TIME_OUT_INT_RAW) {
+		return -ETIMEDOUT;
+	}
+
+	if (wait_cmd) {
+		while (!wait_cmd->done) {
+			r = i2c_esp32_spin_yield(&counter);
+			if (r < 0) {
+				return r;
+			}
+		}
+	}
+
+	/* Wait for I2C bus to finish its business */
+	while (*config->regs.status & I2C_BUS_BUSY) {
+		r = i2c_esp32_spin_yield(&counter);
+		if (r < 0) {
+			return r;
+		}
+	}
+
+	return 0;
+}
+
+static int i2c_esp32_read_msg(struct device *dev, u16_t addr,
+			      struct i2c_msg msg)
+{
+	const struct i2c_esp32_config *config = dev->config->config_info;
+	volatile struct i2c_esp32_cmd *cmd_base = (struct i2c_esp32_cmd *)config->regs.cmd_base;
+	volatile struct i2c_esp32_cmd *cmd_after_write;
+	volatile struct i2c_esp32_cmd *cmd = cmd_base;
+	struct i2c_esp32_data *data = dev->driver_data;
+	u32_t i;
+	int r;
+
+	/* Set the R/W bit to R */
+	addr |= BIT(0);
+
+	i2c_esp32_reset_fifo(config);
+
+	/* Start transaction */
+	if (msg.flags & I2C_MSG_RESTART) {
+		*cmd++ = (struct i2c_esp32_cmd) {
+			.opcode = I2C_ESP32_OP_RSTART
+		};
+	}
+
+	/* Send address */
+	*config->regs.fifo = addr & I2C_FIFO_RDATA;
+	if (data->config.bits.use_10_bit_addr) {
+		*config->regs.fifo = (addr >> 8) & I2C_FIFO_RDATA;
+	}
+	*cmd++ = (struct i2c_esp32_cmd) {
```

Comment:
add an empty line. Basically, always break a line after a '}' unless it's another } etc..

Apply that everywhere (there are some places where it's needed)